### PR TITLE
[ILM] Fix the migrate to tiers service and migrate action tiers configuration

### PR DESCRIPTION
**Pull request by andreidan**
The migrate action (although no allowed in the frozen phase) would seem to convert `frozen` to `data_frozen,data_cold,data_warm,data_hot` tier configuration. As the migrate action is not allowed in the frozen phase this would never happen, however the code is confusing as it seems like it could.

The migrate to data tiers routing service shared the code used by the `migrate` action that converted `frozen` to `data_frozen,data_cold,data_warm,data_hot` if it would encounter an index without any `_tier_preference` setting 
but with a custom node attribute configured to `frozen` e.g. `include.data: frozen`

As part of https://github.com/elastic/elasticsearch/issues/84758 we have seen frozen indices with the `data_frozen,data_cold,data_warm,data_hot` tier preference however we could never reproduce it.

Relates to https://github.com/elastic/elasticsearch/issues/84758

### CODE REVIEW

File: x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/cluster/metadata/MetadataMigrateToDataTiersRoutingServiceTests.java [hunk 1]
```diff
@@ -148,7 +148,7 @@ public void testMigrateIlmPolicyForIndexWithoutILMMetadata() {
         assertThat(migratedColdAllocateAction.getRequire().size(), is(0));
     }
 
-    public void testMigrateIlmPolicyFOrPhaseWithDeactivatedMigrateAction() {
+    public void testMigrateIlmPolicyForPhaseWithDeactivatedMigrateAction() {
```
Comment (andreidan) [inline x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/cluster/metadata/MetadataMigrateToDataTiersRoutingServiceTests.java:151]: Minor typo fix

**Review by gmarouli — APPROVED** [review]
LGTM, thank you for fixing this @andreidan !

**Review by dakrone — APPROVED** [review]
LGTM also, thanks Andrei!

**Comment by dakrone** [comment]
@elasticmachine update branch

**Pull request merged**